src/State/WorldMap/WorldMapEvents.cpp

```diff
@@ -0,0 +1,88 @@
+#include "./WorldMap.h"
+
+namespace Falltergeist
+{
+    namespace State
+    {
+
+        void WorldMap::handle(Event::Event* event)
+        {
+            auto game = Game::getInstance();
+
+            if (auto mouseEvent = dynamic_cast<Event::Mouse*>(event))
+            {
+                auto mouse = game->mouse();
+
+                // Left button down
+                if (mouseEvent->name() == "mousedown" && mouseEvent->leftButton())
+                {
+                    int mouseX = mouse->x();
```

Comment:
`mouseX` and `mouseY` are only used as `unsigned int` below anyway, maybe statically cast them here already?